### ISSUE TITLE
Parameter learning re-enables structurally zero entries in A/B (MINVAL clipping in dirichlet_expected_value)

### ISSUE THREAD

**Issue by conorheins** [infer-actively/pymdp#453]
Parameter learning re-enables entries that are structurally zero in `A`/`B`, and the error compounds across online updates because the returned expected matrices are fed back in as the model.

## Cause

`dirichlet_expected_value` clips before normalizing:

```python
dir_arr = jnp.clip(dir_arr, min=MINVAL)
```

so a cell with zero concentration comes out positive. `update_obs_likelihood_dirichlet` / `update_state_transition_dirichlet` return that as `E_qA` / `E_qB`, `Agent` and `rollout` assign it to `agent.A` / `agent.B`, and the next update accumulates counts there like any other cell.

## Repro

```python
import numpy as np, jax.numpy as jnp
from pymdp.learning import update_obs_likelihood_dirichlet

A = [jnp.eye(2)]          # identity support: A[0][0, 1] is structurally impossible
pA = [jnp.eye(2)]         # zero concentration at the forbidden cells
obs, qs = [jnp.array([1.0, 0.0])[None]], [jnp.array([0.6, 0.4])[None]]

for step in range(3):
    pA, A = update_obs_likelihood_dirichlet(pA, A, obs, qs, A_dependencies=[[0]],
                                            num_obs=[2], lr=1.0, categorical_obs=True)
    print(step, float(np.asarray(A[0])[0, 1]))
```

On `ff1eb8d`:

```
0 0.2857143
1 0.4444444
2 0.5454545
```

The forbidden entry ends up the most likely observation. The `B` path behaves the same way (0.247 after one update in an equivalent setup).

## Relation to #445

#445 adds a `_mask_support` helper that fixes this for the `A` path and for `B` factors that are actually updated. It does not cover the skipped-factor path: with `factors_to_update != "all"`, `update_state_transition_dirichlet` routes untouched factors through the `lax.cond` false branch

```python
lambda: (pB_f, dirichlet_expected_value(pB_f))
```

which never sees the mask. Forbidden cells come back at `MINVAL` (2.2e-16), and since the mask in #445 is derived from the `B` passed in, the next update that *does* touch that factor treats them as supported:

```
update with factors_to_update=[0]   ->  E_qB[1] forbidden cell = 2.2e-16
feed that back in, update [1]       ->  E_qB[1] forbidden cell = 0.247
```

Filing separately so the skipped-factor case does not depend on #445 landing.

## Suggested fix

Keep the support fixed rather than re-deriving it from the running estimate, and apply it on every return path including the `lax.cond` false branch -- or have that branch return the `B_f` it was handed instead of recomputing the expected value. Tests should cover repeated updates with `factors_to_update=[0]`; the current ones only exercise `"all"`.

Also worth noting `support_mask > 0` has no BCOO rule, which matters for the sparse work in #446.
